libs/librrgraph/src/base/rr_graph_storage.cpp

```diff
@@ -354,6 +360,7 @@ void t_rr_graph_storage::assign_first_edges() {
     size_t num_edges = edge_src_node_.size();
     VTR_ASSERT(edge_dest_node_.size() == num_edges);
     VTR_ASSERT(edge_switch_.size() == num_edges);
+    VTR_ASSERT(edge_switch_offset_inf_.size() == num_edges);
```

Comment:
I think one flyweight for edge data would be best, rather than two (switch_inf and edge_switch_offset_inf).  Try to store one flyweight id per edge; we can set it up the same way as today during rr-graph building and then update it if the user requests that with an option saying they want a detailed delay file.